onnxruntime/test/framework/inference_session_test.cc

```diff
@@ -2135,13 +2135,21 @@ TEST(InferenceSessionTests, LoadModelWithEnvVarSetToUnsupportedVal) {
 #endif
 }
 
+struct InferenceSessionExposingSessionState : public InferenceSession {
+  InferenceSessionExposingSessionState(const SessionOptions& session_options,
+                                       const Environment& env)
```

Comment:
side note: we've possibly reached the point of needing a shared helper class derived from InferenceSession that exposes the graph and session state given we do both of those things in multiple places now.